avakas/avakas.py

```diff
@@ -44,32 +44,41 @@ class Avakas():
     """
     project_flavors = {}
 
-    def __init__(self, **kwargs):
+    def __init__(self, directory, tag_prefix='v', **kwargs):
         self._version = Version('0.0.0')
-        self.directory = kwargs['directory'][0]
+        self.tag_prefix = tag_prefix or ''
+        self.directory = directory[0]
         self.options = kwargs
 
     @property
     def version(self):
         """Get version"""
 
-        tag_prefix = self.options.get('tag_prefix', '')
-        return "%s%s" % (tag_prefix, self._version)
+        return f'{self.tag_prefix or ""}{self._version}'
 
     @version.setter
     def version(self, version):
         """Set version"""
-        if not isinstance(version, str):
-            raise TypeError("version must be type of str")
 
-        tag_prefix = self.options.get('tag_prefix', None)
-        version = version.strip(tag_prefix)
+        if not isinstance(version, Version):
 
-        try:
-            self._version = Version(version)
-        except ValueError as err:
-            raise AvakasError("Invalid version string %s" %
-                              version) from err
+            if not version:
+                raise ValueError('Version must non-null/positive length')
+            if not isinstance(version, str):
+                raise TypeError("version must be type of str")
+            if self.tag_prefix and version.startswith(self.tag_prefix):
+                version = version[len(self.tag_prefix):]
+            try:
+                version = Version(version)
+            except ValueError as err:
+                # Doing this to get around the linter, which seems like a
```

Comment:
Which linter target specifically? Were you looking to skip a single line?

Reply:
yeah, let me make the comment more clear, it's the line length, and it's pycodestyle which was having a problem iirc, I was having trouble getting the linter disabled on that single line for line length during the checks. If it's something I've missed that you know of, it was my first preference